tests/document/indexes.py

```diff
@@ -844,7 +844,12 @@ class ReportDictField(Document):
 
         self.assertEqual({'text': 'OK', '_id': {'term': 'ok', 'name': 'n'}},
                          report.to_mongo())
-        self.assertEqual(report, ReportDictField.objects.get(pk=my_key))
+
+        # We can't directly call ReportDictField.objects.get(pk=my_key),
+        # because dicts are unordered, and if the order in MongoDB is
+        # different than the one in `my_key`, this test will fail.
```

Comment:
Example of a test that failed because of this: https://travis-ci.org/MongoEngine/mongoengine/jobs/176644897#L1735